### PR TITLE
[DNM] feat(p2): add index based on std::map

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

as a workaround for those who cannot finish project 2

### CODE REVIEW

**Comment by skyzh** [comment]
This PR should not be merged. Anyone who needs it can pull from this branch.

File: src/include/storage/index/stl_comparator_wrapper.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+namespace bustub {
+
+template <typename KeyType, typename KeyComparator>
+class StlComparatorWrapper {
+ public:
+  explicit StlComparatorWrapper(const KeyComparator &cmp) : cmp_{cmp} {}
+
+  inline auto operator()(const KeyType &lhs, const KeyType &rhs) const -> bool { return cmp_(lhs, rhs) < 0; }
```
Comment (lmwnshn) [inline src/include/storage/index/stl_comparator_wrapper.h:10]: Oh! I just realized why my wrapper on Friday was seeing weird bugs...

I thought `operator()` would return `int` like a Java comparator, not a `bool` based on `std::less`. So I just used `cmp_(lhs, rhs)` directly. Oops.

Comment (skyzh) [reply src/include/storage/index/stl_comparator_wrapper.h:10]: Yeah, I also feel C++'s compile error is unhelpful and cannot point to the exact place that I got wrong with 😇

**Comment by sacloudy** [comment]
Does the autograder server contain this header file? When I pull this branch and submit it to autograder, I got this error: 
/autograder/source/bustub/src/include/storage/index/index_iterator.h:20:10: fatal error: 'storage/index/stl_comparator_wrapper.h' file not found

**Comment by skyzh** [comment]
> Does the autograder server contain this header file? When I pull this branch and submit it to autograder, I got this error:
> /autograder/source/bustub/src/include/storage/index/index_iterator.h:20:10: fatal error: 'storage/index/stl_comparator_wrapper.h' file not found

See the Discord discussion.

**Pull request closed without merging**